### ISSUE TITLE
References to the parent field of a slot arent duplicated/saved correctly

### ISSUE THREAD

**Issue by LucasRo7** [Frooxius/NeosPublic#947]
From discord(with little modifications):
So, basically, if you take the logix interface of a slot(inside the "object"), connect a dereference node ("->") from the "parent" of the slot to a display, its gonna show the parent of the object(for example, the root)

If you pack the logix into the object and unpack it(so that the logix is parented under the object) and duplicate the object(or save it and spawn it out), the display node shows "null" and inspecting the arrow node(the _ReferenceNode_) shows that its ref target also set to null(hence why they loose the reference)

The same thing happens when you make a _ReferenceField_(attach component->data->ReferenceField) of type "FrooxEngine.SyncRef`1[FrooxEngine.Slot]" and drag+drop the parent onto the reference field, after duplicating the slot, the ReferenceField becomes null. 
This does not happen on other slot fields: if you use the target of a _ReferenceField\<Slot>_, the target from the logix Slot variable or bones on a skinned mesh it will duplicate and the copy wont be set to null

This behaciour also happens if you use other means of storing the reference, for example: 
- the _IValue\<RefID>_, the _ReferenceNode\<Slot>_ and _DriverNode\<RefID>_ logix nodes
- _ValueCopy\<RefID>_ and _ReferenceCopy\<Slot>_(both source and target) 
- _ReferenceField\<Sync\<RefID>>_, _BooleanValueDriver\<RefID>_ and _ValueEqualityDriver\<Slot>_
- _RefEditor_, which is used by inspectors to modify the parent

Interestingly, grabbing the object and switching worlds keeps the references

**TLDR:** any references to the parent field of a slot(that are also under it) are set to null whenever the slot or one of its parents are duplicated or saved and spawned

**Comment by JackTheFoxOtter** [comment]
Stumbled accross this today as well, this bug could potentially break working items on duplication if the item utilizes references to a slot's parent field in its internal functionality without the user / creator realizing at first. Local references are expected to not break on duplication.
